### PR TITLE
Add conversation resume and start control to WebChat adapter

### PR DESCRIPTION
**Pull request by adilei**
## Summary

Adds two new settings to `CopilotStudioWebChat.createConnection()` that enable resuming existing conversations and controlling the initial greeting behavior:

- **`conversationId`** — pass an existing conversation ID to resume a previous conversation instead of starting a new one. When provided, `startConversationStreaming()` is skipped by default and subsequent `postActivity` calls route to the existing conversation.
- **`startConversation`** — explicitly control whether `startConversationStreaming()` is called on subscribe. Defaults to `!conversationId` (skip when resuming, call when new).

Also exposes a `conversationId` getter on the returned connection object, allowing callers to capture and persist the active conversation ID for later resume.

## Motivation

Web applications that embed Copilot Studio via WebChat often need to preserve conversation continuity across page navigations. Without this change, every `createConnection()` call starts a fresh conversation, losing all prior context.

## Changes

**`copilotStudioWebChat.ts`**
- `CopilotStudioWebChatSettings`: added `conversationId?: string` and `startConversation?: boolean`
- `CopilotStudioWebChatConnection`: added `readonly conversationId: string | undefined`
- `createConnection()`: tracks `activeConversationId`, passes it to `sendActivityStreaming()`, adds `started`/`ended` guards to prevent duplicate starts and post-end sends

**`copilotStudioWebChat.test.ts`** (new)
- 15 tests covering default behavior, conversation resume, start control, error handling, and edge cases

## Usage

```typescript
// New conversation (unchanged behavior)
const connection = CopilotStudioWebChat.createConnection(client);

// Resume existing conversation
const connection = CopilotStudioWebChat.createConnection(client, {
  conversationId: savedConversationId,
});

// Capture conversationId for persistence
connection.activity$.subscribe(() => {
  if (connection.conversationId) {
    localStorage.setItem('conversationId', connection.conversationId);
  }
});
```

## Test plan

- [x] All 15 new tests pass
- [x] All 32 existing `agents-copilotstudio-client` tests pass (47/47 total)

### CODE REVIEW

File: packages/agents-copilotstudio-client/src/copilotStudioWebChat.ts [hunk 1]
```diff
@@ -224,9 +257,15 @@ export class CopilotStudioWebChat {
     settings?: CopilotStudioWebChatSettings
   ): CopilotStudioWebChatConnection {
     logger.info('--> Creating connection between Copilot Studio and WebChat ...')
+
+    const shouldStart = settings?.startConversation ?? !settings?.conversationId
+
     let sequence = 0
     let activitySubscriber: Subscriber<Partial<Activity>> | undefined
     let conversation: ConversationAccount | undefined
+    let activeConversationId: string | undefined = settings?.conversationId
```
[the remark this answers is not in the document]
Comment (adilei) [reply packages/agents-copilotstudio-client/src/copilotStudioWebChat.ts]: Already addressed — the current code normalizes the conversationId by trimming whitespace and treating empty strings as `undefined` before computing `shouldStart`:

```ts
const normalizedConversationId =
  settings?.conversationId && settings.conversationId.trim() !== ''
    ? settings.conversationId.trim()
    : undefined
const shouldStart = settings?.startConversation ?? !normalizedConversationId
```
